MultiSource/Benchmarks/Ptrdist/anagram/anagram.c

```diff
@@ -223,24 +226,32 @@ typedef struct {
 } Letter;
 typedef Letter * PLetter;
 
-Letter alPhrase[ALPHABET]; /* statistics on the current phrase */
+// CHECKEDC : checkedc array type
+// checkedc array type of T == array pointer type of T
+// bounds information for checkedc array is from array size
+Letter alPhrase _Checked [ALPHABET]; /* statistics on the current phrase */
 #define lPhrase(ch) alPhrase[ch2i(ch)]  /* quick access to a letter */
 
 int cchPhraseLength;                    /* number of letters in phrase */
 
-Quad aqMainMask[MAX_QUADS];/* the bit field for the full phrase */
-Quad aqMainSign[MAX_QUADS];/* where the sign bits are */
+Quad aqMainMask _Checked [MAX_QUADS];/* the bit field for the full phrase */
+Quad aqMainSign _Checked [MAX_QUADS];/* where the sign bits are */
+
+char achPhrase _Checked [255];
 
 int cchMinLength = 3;
 
 /* auGlobalFrequency counts the number of times each letter appears, summed
  * over all candidate words.  This is used to decide which letter to attack
  * first.
  */
-unsigned auGlobalFrequency[ALPHABET];
-char achByFrequency[ALPHABET];          /* for sorting */
+unsigned auGlobalFrequency _Checked [ALPHABET];
+char achByFrequency _Checked [ALPHABET];          /* for sorting */
 
-char * pchDictionary;               /* the dictionary is read here */
+// CHECKEDC : externally scoped variable bounds information
+// global variable bounds information is defined in declaration
+// dictionary maximum length is defined as MAXWORDS
+_Array_ptr<char> pchDictionary : count(MAXWORDS);               /* the dictionary is read here */
```

Comment:
pchDictionarySize is not yet determined in declaration
In somewhere of codes, pchDictionarySize will be determined and pchDictionary is also allocated
as follows:

```
pchDictionarySize = ..... 
pchDictionary = malloc(pchDictionarySize)
```
When assigning pchDictionSize value, I have understood that the bounds for pchDictionary is invalided since pchDictionarySize is used to represent bounds info of pchDictionary pointer variable
In specification (ch3.2.4), 

> Because externall scoped array_ptr variables can have bounds declared for them only at their definitions, by definition their bounds are always invariants 

Does it work?